### DISCUSSION TITLE
Authentication support

### DISCUSSION THREAD

**Discussion by dometto** [gollum/gollum#2088] [Ideas]
Third party authentication gems for Gollum like https://github.com/bjoernalbers/gollum-auth and https://github.com/arr2036/omnigollum/issues seem not to be very actively maintained anymore. Since authentication is a pretty common demand, I'm wondering how we can best address this lacuna. I see two general options:

1. Fork and start maintaining one of the authentication extensions as part of the gollum org, allowing us to also publish Docker images with auth configured.
2. Don't rely on a gem for authentication at all, instead provide some `docker-compose` examples showing how to achieve auth using e.g. a separate nginx container.

Given the additional maintenance burden involved in 1., my preference would go to 2. What do others think?

Some examples we could provide (feel free to suggest more/others):

* [CAS](https://hub.docker.com/r/apereo/cas) in Docker
* [Traefik](https://github.com/JensKnipper/traefik-examples/blob/master/authentication/basic-authentication/docker-compose.yml) with basic auth in Docker

**Comment by vadz** [comment]
From my (user) point of view, integrating gollum-auth into gollum would be the best. It's a pretty simple gem, so the cost/benefit ratio for just integrating it directly seems to be pretty low.
